test/python/transpiler/test_preset_passmanagers.py

```diff
@@ -919,3 +919,32 @@ def test_input_dag_copy(self):
         qc.cx(1, 0)
         circ = transpile(qc, basis_gates=["u3", "cz"])
         self.assertIsInstance(circ, QuantumCircuit)
+
+
+@ddt
+class TestOptimizationOnSize(QiskitTestCase):
+    """Test the optimization levels for optimization based on
+    both size and depth of the circuit.
+    See https://github.com/Qiskit/qiskit-terra/pull/7542
+    """
+
+    @data(1, 2, 3)
+    def test_size_optimization(self, level):
+        """Test the levels for optimization based on size of circuit"""
+        qc = QuantumCircuit(8)
+        qc.cx(5, 4)
+        qc.cx(6, 5)
+        qc.cx(4, 5)
+        qc.cx(5, 6)
+        qc.cx(5, 4)
+        qc.cx(6, 7)
+        qc.cx(6, 5)
+        qc.cx(5, 4)
+        qc.cx(7, 6)
+        qc.cx(6, 7)
+
+        circ = transpile(qc, optimization_level=level)
+
+        self.assertDictEqual(qc.count_ops(), circ.count_ops())
+        self.assertEqual(qc.size(), circ.size())
+        self.assertEqual(qc.depth(), circ.depth())
```

Comment:
It looks like these are asserting that the input and output of the call to `transpile` have the same `count_ops()`, `size()` and `depth()`, but I'm not sure that's sufficient to make sure we're covering the case from #7542 . Maybe we can use the input there as a test case, and verify that the size of the transpiled circuit is lower than its input (though I think this would only happen at `optimization_level={2,3}`).

Reply:
Yes, I agree that even after adding the check for the size, there wasn't any optimization in level 1. I think the optimization routine for level 1 does not include the correct transformation pass for this reduction to occur (not sure about the name)